### PR TITLE
Improves and fixes inconsistencies in boundary conditions docs

### PR DESCRIPTION
**Pull request by glwagner**

### CODE REVIEW

File: docs/src/model_setup/boundary_conditions.md [hunk 1]
```diff
@@ -9,44 +9,49 @@ y-dimension, left and right are called south and north. For the z-dimension, lef
...
+   _Note_: You cannot use `ValueBoundaryCondition` on a wall-normal velocity component; you must use `Open` for that.
+   Examples where you might use `ValueBoundaryCondition`:
+   * Prescribe a surface to have a constant temperature, like 20 degrees.
+     Heat will then flux in and out of the domain depending on the temperature difference between the surface and the interior, and the temperature diffusivity.
+   * Prescribe a velocity tangent to a boundary as in a driven-cavity flow (for example), where the top boundary is moving.
+     Momentum will flux into the domain do the difference between the top boundary velocity and the interior velocity, and the prescribed viscosity.
 
-Oceananigans uses a hierarchical structure to express boundary conditions.
+4. `FluxBoundaryCondition` _directly_ prescribes the flux of a quantity across a boundary rather than calculating it given a viscosity or diffusivity.
+   For example, sunlight absorbed at the ocean surface imparts a temperature flux that heats near surface fluid.
```
Comment (navidcy) [inline docs/src/model_setup/boundary_conditions.md:44]:
```suggestion
   For example, sunlight absorbed at the ocean surface imparts a temperature flux that heats near-surface fluid.
```

File: docs/src/model_setup/boundary_conditions.md [hunk 2]
```diff
@@ -9,44 +9,49 @@ y-dimension, left and right are called south and north. For the z-dimension, lef
...
+   If you have a known diffusivity, you can express `FluxBoundaryCondition(flux)` using `GradientBoundaryCondition(-flux / diffusivity)` (aka "Neumann" boundary condition).
```
Comment (navidcy) [inline docs/src/model_setup/boundary_conditions.md:45]:
```suggestion
   If there is a known `diffusivity`, you can express `FluxBoundaryCondition(flux)` using `GradientBoundaryCondition(-flux / diffusivity)` (aka "Neumann" boundary condition).
```

**Review by navidcy — APPROVED** [review]
Let's wait to see the Docs preview first!

**Comment by navidcy** [comment]
how about changing `!!! info "The flux convention in Oceananigans"` to `!!! warning "The flux convention in Oceananigans"`?

**Comment by glwagner** [comment]
> how about changing `!!! info "The flux convention in Oceananigans"` to `!!! warning "The flux convention in Oceananigans"`?

Ahaha I'm not sure it qualifies as a "warning" does it? Nothing is the matter, it's just info you need to know!

**Comment by glwagner** [comment]
Preview here: https://clima.github.io/OceananigansDocumentation/previews/PR1836/model_setup/boundary_conditions/

**Comment by navidcy** [comment]
@glwagner, shall I/u merge?

**Comment by navidcy** [comment]
I'm taking the liberty...

**Comment by glwagner** [comment]
Ah, was just waiting for approval that you were ok I didn't add the warning you requested!

**Comment by navidcy** [comment]
That was just a suggestion, not a requirement. ;)

**Pull request merged**